src/layer/Tooltip.js

```diff
@@ -413,9 +413,22 @@ Layer.include({
 
 
 	_openTooltip(e) {
-		if (!this._tooltip || !this._map || (this._map.dragging && this._map.dragging.moving())) {
+		if (!this._tooltip || !this._map) {
 			return;
 		}
+
+		// If the map is moving, we will show the tooltip after it's done.
+		if (this._map.dragging && this._map.dragging.moving()) {
+			if (!this._openOnceFlag) {
+				this._openOnceFlag = true;
+				this._map.once('moveend', () => {
+					this._openOnceFlag = false;
+					this._openTooltip(e);
+				});
+			}
```

Comment:
@mourner @jonkoops @IvanSanchez have someone a good Idea to make this cleaner? The `once` listener should not be added multiple times and it is important to pass `e` as argument from the original `_openTooltip` call.

Reply:
Here's a couple of ideas:

Superficially, you could do something like this:

```
                // If the map is moving, we will show the tooltip after it's done.
                if (this._map.dragging && this._map.dragging.moving() && !this._openOnceFlag) {
                        this._openOnceFlag = true;
                        this._map.once('moveend', () => {
                            this._openOnceFlag = false;
                            this._openTooltip(e);
                        });
                        return;
                }
```

It's not satisfying because we still have to store `_openOnceFlag`.

However, maybe there's a better solution.  There are only four places where `_openTooltip` is called at all, and they're all here in `Tooltip.js`.

We could replace all instances of `_openTooltip` with `_openTooltipWhenNotMoving`, with:

```
_openTooltipWhenNotMoving(e) {
    if (this._map.dragging && this._map.dragging.moving()) {
        this._map.once('moveend', () => {
            this._openTooltip(e);
        });
        return;
    } else {
        this._openTooltip(e);
    }
}
```

Then remove the moving checks from `_openTooltip` proper.

In this way, we avoid storing an extra variable, and call `_openTooltip` at most once at an appropriate time when it would be needed.

Reply:
I've implemented the former solution, which at least resolves a little bit of the nesting.

I don't see a better solution than the `_openOnceFlag` that you implemented, and I've convinced myself that it is correct even in case where some external code toggles `permanent`.